definitions/features/candlepin_database.rb

```diff
@@ -35,9 +35,7 @@ def check_option_using_cpdb_help(option_name, parent_cmd = '')
   def execute_cpdb_validate_cmd
     main_cmd = cpdb_validate_cmd
     return [true, nil] if main_cmd.empty?
-    main_cmd += format_shell_args(
-      '-u' => configuration['username'], '-p' => configuration[%(password)]
-    )
+    main_cmd += " -u #{configuration['username']} -p #{configuration[%(password)]} "
```

Comment:
I think the real problem is in 
https://github.com/theforeman/foreman_maintain/blob/494a3f12955bbb1b9f453d463a3b50d88c969175/lib/foreman_maintain/concerns/system_helpers.rb#L149-L151

You use `Shellwords.shellescape`, but then add single quotes around the command.
However, the [docs for `shellescape` say](https://ruby-doc.org/stdlib-2.5.1/libdoc/shellwords/rdoc/Shellwords.html#method-c-shellescape): "Note that a resulted string should be used unquoted and is not intended for use in double quotes nor in single quotes."